.github/workflows/build-and-release-maven.yml

```diff
@@ -0,0 +1,57 @@
+name: Build and Release Maven project
+
+on:
+  workflow_call:
+    inputs:
+      skip-tests:
+        description: skips the tests during the Build phase if set to 'true'
+        type: boolean
+        required: false
+        default: false
+    secrets:
+        BOT_GITHUB_TOKEN:
+          required: true
+        NEXUS_USERNAME:
+          required: true
+        NEXUS_PASSWORD:
+          required: true
+
+env:
+  GIT_PASSWORD: ${{ secrets.BOT_GITHUB_TOKEN }}
+  MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+  MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+
+jobs:
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.32.0
+      - name: "Build"
+        if: ${{ inputs.skip-tests }}
+        run: mvn -B -V install -DskipTests
+      - name: "Build and Test"
+        if: ${{ !inputs.skip-tests }}
+        run: mvn -B -V install
+
+  release:
+    name: "Release"
+    runs-on: ubuntu-latest
+    needs: build
+    if: >
+      github.ref_name == 'master' && github.event_name != 'pull_request' &&
+       !contains(github.event.head_commit.message, '[no-release]')
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.32.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/configure-git-author@v1.32.0
+        with:
+          username: ${{ secrets.BOT_GITHUB_USERNAME }}
+          email: ${{ secrets.BOT_GITHUB_EMAIL }}
```

Comment:
this should be added in `secrets` as well, otherwise even better should be moved to inputs since they are not really sensitive.

The second option is going to add more code in the workflow of each repository but would make this action really reusable outside the alfresco org, I think it's worth it.

Reply:
Thanks for the feedback, given that it will be used only within the Alfresco org and `BOT_GITHUB_USERNAME` and `BOT_GITHUB_EMAIL` are both set as secrets for this org, I'd rather keep them as required secrets with no additional impact on currently migrated Alfresco workflows that would come from transitioning them to inputs instead.

If they'll ever be changed from `secrets` to `env` vars I'll be happy to refactor this though 👍